src/restconfig/src/main/java/org/geoserver/catalog/rest/CoverageResource.java

```diff
@@ -150,8 +151,9 @@ protected void handleObjectPut(Object object) throws Exception {
         
         CoverageStoreInfo cs = catalog.getCoverageStoreByName(workspace, coveragestore);
         CoverageInfo original = catalog.getCoverageByCoverageStore( cs,  coverage );
-        new CatalogBuilder(catalog).updateCoverage(original,c);
         calculateOptionalFields(c, original);
+        
+        new CatalogBuilder(catalog).updateCoverage(original,c);
```

Comment:
calculateOptionalFields itself also got heavily overhauled (see above). The method is part of the restconfig resource implementation, and appears to have origionally been written primarily with REST transactions in mind. See https://github.com/geoserver/geoserver/commit/558e4bad12ecdb596ead3a695cfabc8f7ee162a1 for more info.